### ISSUE TITLE
Turtle mined blocks missing in turtle inventory

### ISSUE THREAD

**Issue by oskarbukovsky** [SquidDev-CC/CC-Tweaked#1968]
### Minecraft Version

1.20.1

### Version

1.113.0

### Details

We registered that mining turtles stopped mining properly some time ago, from what i was looking, last working version was 1.110.3, from 1.111 up turtle digs blocks as normal, but nothing is stored in inventory, no log errors, nothing. Idk if there is any way to look for things that could cause that, because its some time and i dont have backup of server from that time, we use plugins with  some (365 mods), so its looking in haybale. But things like grinders, spikes, ae2 panels which place and destroy blocks works fine.

**Comment by SquidDev** [comment]
Thanks for the report. I'm afraid I'm unable to reproduce this (Forge 47.1.0, CC:T 1.113.0).

Looking at the diff between [1.110.3 and 1.111.0](https://github.com/cc-tweaked/CC-Tweaked/compare/v1.20.1-1.110.3..v1.20.1-1.111.0), I don't see any changes that would affect turtle breaking behaviour. Are you sure you didn't update any other mods at the same time?

Would you be able to upload your logs? It's always useful to include these, even if they don't include any errors, as it contain lots of other useful information like mod versions!

**Comment by SquidDev** [comment]
> we use plugins 

I'd missed this the first time round, but judging by other issues you've opened, it looks like you're running Mohist. This is almost definitely an issue on their end.

I'd strongly recommend against running any of the Bukkit-on-Forge projects. The two systems do not play well together, and invariably cause all sorts of problems. You're much better off trying to find native Forge replacements for your various plugins.

**Comment by ArsiTheFox** [comment]
> Thanks for the report. I'm afraid I'm unable to reproduce this (Forge 47.1.0, CC:T 1.113.0).
> 
> Looking at the diff between [1.110.3 and 1.111.0](https://github.com/cc-tweaked/CC-Tweaked/compare/v1.20.1-1.110.3..v1.20.1-1.111.0), I don't see any changes that would affect turtle breaking behaviour. Are you sure you didn't update any other mods at the same time?
> 
> Would you be able to upload your logs? It's always useful to include these, even if they don't include any errors, as it contain lots of other useful information like mod versions!

If they are using ATM9 I am also having this issue on our server... and I believe I know the issue.... This pack has 2 turtles in custom village structures and I believe because of that there is lots of Turtle ID conflicts between player placed turtles and world placed. In the pack I cannot use a turtle to mine for me because it doesn't pickup any of the blocks mined.

I don't know how the mod works in the background but having structures with turtles in them caused wackyness with turtle ids

**Comment by fatboychummy** [comment]
The turtles in structures shouldn't cause issues. IDs are not assigned until the turtle turns on for the first time -- unless the mod author made the mistake of putting turtles with an ID in the structure, which I'm fairly sure was an issue that got fixed there ages ago!

It also shouldn't matter if their IDs conflict, each turtle should have their own inventory regardless of their ID. I have messed around with cloned turtles in creative mode for testing programs in the past and never ran into an issue with inventories being incorrect.
